unity-renderer/Assets/Rendering/Culling/CullingObjectsTracker.cs

```diff
@@ -1,144 +1,198 @@
     public class CullingObjectsTracker : ICullingObjectsTracker
     {
-        private ICollection<Renderer> renderers;
-        private ICollection<SkinnedMeshRenderer> skinnedRenderers;
-        private Animation[] animations;
+        private readonly List<Rendereable> rendereables = new List<Rendereable>();
+        private ICollection<Renderer> usingRenderers = new List<Renderer>();
+        private ICollection<SkinnedMeshRenderer> detectedSkinnedRenderers;
+        private ICollection<SkinnedMeshRenderer> usingSkinnedRenderers = new List<SkinnedMeshRenderer>();
+        private Animation[] usingAnimations;
 
         private int ignoredLayersMask;
         private bool dirty = true;
-
+        
 
         public CullingObjectsTracker()
         {
             PoolManager.i.OnGet -= MarkDirty;
             PoolManager.i.OnGet += MarkDirty;
+
+            Rendereable.OnRendereableAdded -= OnRendereableAdded;
+            Rendereable.OnRendereableAdded += OnRendereableAdded;
         }
 
-        /// <summary>
-        /// If the dirty flag is true, this coroutine will re-populate all the tracked objects.
-        /// </summary>
+        /// <summary> Adds a rendereable to be tracked and calculating in the culling process. </summary>
+        public void OnRendereableAdded(object sender, Rendereable rendereable) => rendereables.Add(rendereable);
+
+        /// <summary> Sets a layer mask to be ignored, only one layer mask can be ingored at a given time. </summary>
+        public void SetIgnoredLayersMask(int ignoredLayersMask) { this.ignoredLayersMask = ignoredLayersMask; }
+
+        /// <summary> This will re-populate all the tracked objects synchronously </summary>
+        public void ForcePopulateRenderersList()
+        {
+            detectedSkinnedRenderers = Object.FindObjectsOfType<SkinnedMeshRenderer>();
+
+            ForceCalculateRenderers();
+            ForceCalculateSkinnedRenderers();
+            usingAnimations = Object.FindObjectsOfType<Animation>();
+
+            dirty = false;
+        }
+
+        /// <summary> If the dirty flag is true, this coroutine will re-populate all the tracked objects. </summary>
         public IEnumerator PopulateRenderersList()
         {
             if (!dirty)
                 yield break;
 
-            List<Renderer> renderersList = new List<Renderer>();
-            List<SkinnedMeshRenderer> skinnedRenderersList = new List<SkinnedMeshRenderer>();
-            Renderer[] allRenderers = Object.FindObjectsOfType<Renderer>();
-
-            // TODO new task, avoid FindGameObjectsOfType and use:
-            // SceneManager.GetActiveScene().GetRootGameObjects(), calculate from there.
+            detectedSkinnedRenderers = Object.FindObjectsOfType<SkinnedMeshRenderer>();
 
+            yield return null;
+            yield return CalculateRenderers();
+            yield return null;
+            yield return CalculateSkinnedRenderers();
+            yield return null;
```

Comment:
Why do we need so many frame skips, what is the purpose?

Reply:
This enforces that no more than "n" elements are calculated in a single frame